### ISSUE TITLE
`navigator is not defined` with Angular Universal

### ISSUE THREAD

**Issue by jtammen** [dbfannin/ngx-logger#136]
Hey,

I am trying to use newest version of ngx-logger (4.0.3) in an Angular Universal project. Angular CLI: 8.1.2, Node: 12.3.1.

This is the error I get when trying to open the app:

```
ReferenceError: navigator is not defined
    at new NGXLogger (dist/server.js:210176:22)
    at _createClass (dist/server.js:24027:20)
```

It works fine in 3.3.13.

**Comment by svera-co** [comment]
This is still an issue with ngx-logger 4.0.4 and Angular 8.2.6 Universal

**Comment by C0ZEN** [comment]
I have the same problem, did you find a solution please ?

**Comment by jtammen** [comment]
> I have the same problem, did you find a solution @jtammen please ?

@C0ZEN No, not yet, I just stayed on version 3.3.13 for now :-/

**Comment by DaSchTour** [comment]
I created a fix reintroducing the isPlatform that was available previously #158

**Comment by DaSchTour** [comment]
There is a bit hacky workaround for this to overwrite NGXLogger on server side. It's ugly due to missing interface for NGXLogger that allows a custom implementation of logger.
```
import { isPlatformBrowser } from '@angular/common';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { FactoryProvider, Inject, PLATFORM_ID } from '@angular/core';
import { LoggerConfig, NGXLogger, NGXLoggerHttpService, NGXLoggerMonitor, NGXMapperService } from 'ngx-logger';

// @ts-ignore
class FallbackLogger implements NGXLogger {
  private _customHttpHeaders: any;
  private _customParams: any;
  private readonly _isIE: any;
  private _log: any;
  private readonly _logFunc: any;
  private _logIE: any;
  private _logModern: any;
  private _loggerMonitor: any;
  private _withCredentials: any;
  private config: any;
  private readonly httpService: any;
  private readonly mapperService: any;

  public debug(message: any, ...additional: Array<any>): void {
    console.debug(message, ...additional);
  }

  public error(message: any, ...additional: Array<any>): void {
    console.error(message, ...additional);
  }

  public fatal(message: any, ...additional: Array<any>): void {
    console.error(message, ...additional);
  }

  public getConfigSnapshot(): LoggerConfig {
    return undefined;
  }

  public info(message: any, ...additional: Array<any>): void {
    console.info(message, ...additional);
  }

  public log(message: any, ...additional: Array<any>): void {
    console.log(message, ...additional);
  }

  public registerMonitor(monitor: NGXLoggerMonitor): void {}

  public setCustomHttpHeaders(headers: HttpHeaders): void {}

  public setCustomParams(params: HttpParams): void {}

  public setWithCredentialsOptionValue(withCredentials: boolean): void {}

  public trace(message: any, ...additional: Array<any>): void {}

  public updateConfig(config: LoggerConfig): void {}

  public warn(message: any, ...additional: Array<any>): void {
    console.warn(message, ...additional);
  }
}

const loggerFactory = (
  platformId: string,
  mapperService: NGXMapperService,
  httpService: NGXLoggerHttpService,
  loggerConfig: LoggerConfig,
) => {
  if (isPlatformBrowser(platformId)) {
    return new NGXLogger(mapperService, httpService, loggerConfig);
  } else {
    return new FallbackLogger();
  }
};

export const loggerProvider: FactoryProvider = {
  provide: NGXLogger,
  useFactory: loggerFactory,
  deps: [[new Inject(PLATFORM_ID), PLATFORM_ID], NGXMapperService, NGXLoggerHttpService, LoggerConfig],
};
```

Now you can add `loggerProvider` to providers in app.module

**Comment by dbfannin** [comment]
This should be fixed in ngx-logger@4.0.8
